packages/dashboard/src/managers/resource-manager-tasks.ts

```diff
@@ -1,14 +1,29 @@
-import { TaskDefinition } from 'react-components';
+import { getTaskDefinition, TaskDefinition } from 'react-components';
+
+export interface TaskResource {
+  task_definition_id: string;
+  display_name?: string;
+}
 
 export class TaskResourceManager {
-  supportedTasks: Record<string, TaskDefinition>;
+  tasks: TaskDefinition[];
 
-  constructor(supportedTasks: TaskDefinition[] | undefined) {
-    this.supportedTasks = {};
-    if (supportedTasks) {
-      for (const t of supportedTasks) {
-        this.supportedTasks[t.taskDefinitionId] = t;
+  constructor(taskResources: TaskResource[]) {
+    this.tasks = [];
+    for (const taskResource of taskResources) {
+      const definition = getTaskDefinition(taskResource.task_definition_id);
+      if (!definition) {
+        console.error(
+          `Invalid tasks configured for dashboard: [${taskResource.task_definition_id}]`,
+        );
+        continue;
       }
+
+      if (taskResource.display_name !== undefined) {
+        definition.taskDisplayName = taskResource.display_name;
```

Comment:
Modifying global const is usually is a code smell. Have a function to get the display name instead.

Reply:
refactored it in https://github.com/open-rmf/rmf-web/pull/925/commits/ba6be3062714389593027cdac565d0e593027c1b, still not using a function as there will be 2 sources of truth, I would rather the resource manager handles all the configuration and massaging of definitions.